packages/pilot/src/pages/Balance/Balance.js

```diff
@@ -322,33 +345,60 @@ class Balance extends Component {
       && company.default_recipient_id
     ) {
       const pathId = getValidId(getRecipientId(company), null)
-      this.updateQuery(this.state.query, pathId)
+
+      this.updateQuery(this.getQuery(), pathId)
     }
   }
 
+  getQuery (newTimeframe) {
+    const {
+      history: {
+        location: {
+          search,
+        },
+      },
+    } = this.props
+    const timeframe = getTimeframeProp(qs.parse(search))
+    const { query } = this.state
+
+    return query[newTimeframe || timeframe]
+  }
+
+  setQuery (query) {
+    const { query: stateQuery } = this.state
+    const timeframe = getTimeframeProp(query)
+    const newQuery = assoc(timeframe, query, stateQuery)
+    this.setState({
```

Comment:
será que rola um :tada: `useState` :tada: aqui?
```suggestion
    const { query: stateQuery } = useState({
      query: newQuery
    })
```

Reply:
Infelizmente não podemos misturar `hooks`  com `class components`, mas gostei muito da ideia, nos próximos componentes quer fizermos com certeza usaremos `hooks`.